tests/utils.py

```diff
@@ -0,0 +1,32 @@
+from random import choice
+from abelian.groups import LCA
+
+
+def random_zero_heavy(low, high):
+    """
+    Draw a random number, with approx 50% probability of zero.
+    """
+    return choice(list(range(low, high)) + [0] * (high - low))
+
+
+def random_from_list(number, list_to_take_from):
```

Comment:
Isnt this more or less the same as

https://docs.python.org/3/library/random.html#random.choices

?

Reply:
I didn't look in details into these methods. I just put the method in the script utils.py

But I think you could write something like (if you want to use `random.choices`):

```python
def generate_random_list(length, low, high):
    choices = [0, random.randint(low, high+1)] 
    random_list = random.choices(choices, k=length)
    
    return random_list
```

Reply:
My bad, I thought it did exactly the same as random.choices. If not, you can keep your implmentation, or use the one above - either way is good :)